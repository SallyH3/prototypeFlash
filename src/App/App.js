import React, { Component } from 'react';
import '../Styles/Main.scss';
import Header from '../Header/Header.js';
import FilteredControls from '../FilteredControls/FilteredControls.js';
import Cards from '../Cards/Cards.js';
import Data from '../Data.js';

export default class App extends Component {
  constructor() {
    super();
    this.state={
      cards:[],
      mutators: [],
      iterations: [],
      accessors: [],
      currentCard:null,
      currentSelection: '',
      getLink: Data.map(method => {
        return {link: method.link, id: method.id}
      })
    }
  }

  setCards = (event)=>{
    let cards = Data.filter((method) => {
      return method.title === event.target.id;
    })
    this.setState({cards}, this.getRandomNumber)
  }

  //these 3 should be one function
// getMutatorMethods = () => {
//   this.setState({currentSelection: 'Mutator methods'},this.setCards)
// }

// getIterationMethods = () => {
//   this.setState({currentSelection: 'Iteration methods'},this.setCards)
// }

// getAccessorMethods = () => {
//   this.setState({currentSelection: 'Accessor methods'},this.setCards)
// }
//rename randomizer
getRandomNumber = ()=>{
  let cards = this.state.cards
const cardIndex =   Math.floor(Math.random() * Math.floor(cards.length - 1));
this.setState({
  currentCard:cards[cardIndex]
})
}

checkReturnCard=()=>{
  const currentCard = this.state.currentCard
  console.log(currentCard)
  if(!currentCard){
    return <div></div>
  }else{
    const answer = currentCard.answer
    const question = currentCard.question
    const link = currentCard.link
  return (<Cards 
    answer={answer}
    question={question}
    link={link}
    getRandomNumber={this.getRandomNumber}
  />)
}  
}

  render() {
    return (
      <section className='app'>
        <Header />
        <FilteredControls 
          mutator={this.setCards}
          iteration={this.setCards}
          accessor={this.setCards}
        />
        {this.checkReturnCard()}
      </section>
    );
  }
}

