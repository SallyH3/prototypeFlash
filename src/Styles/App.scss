@import 'Mixins';
*,
*:after,
*:before {
  box-sizing: border-box;
  font-weight: lighter;
}

.app {
  display: flex;
}

body {
  background-color: $lavender;
  color: $main-color;
}


