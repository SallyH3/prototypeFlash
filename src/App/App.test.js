import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

const mockLink = [];

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <App getLink={mockLink}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default states', () => {
  expect(wrapper.state()).toEqual({
    cards:[],
    mutators: [],
    iterations: [],
    accessors: [],
    deckInUse: [],
    currentCard: null,
    currentSelection: ''
});
});
});