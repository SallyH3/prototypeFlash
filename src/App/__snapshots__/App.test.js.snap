// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<section
  className="app"
>
  <Header />
  <FilteredControls
    accessor={[Function]}
    iteration={[Function]}
    mutator={[Function]}
  />
  <div />
</section>
`;
